shotover-proxy/src/message/mod.rs

```diff
@@ -72,6 +76,32 @@ impl Message {
             original: RawFrame::None,
         }
     }
+
+    pub fn to_filtered_reply(&self) -> Message {
```

Comment:
To me this is leaking a transform specific implementation into `Message`. We won't be using `to_filtered_reply` anywhere but the `QueryTypeFilter` transform so I think it should be there.

Reply:
reasons for putting it in Message:
* Any transform that filters out messages can make use of it.
* The transform doesnt have to manually deal with each protocol type.

But I dont mind too much either way.